### PR TITLE
implement "leightweight" export with external dependencies

### LINKED ISSUE #672: [package.json] Iterate Exported Files
When package.json is present and the viz is exported, there are currently a few issues.

 * The `index.html` does not include the `bundle.js` build at all, so it does not actually run unless the user adds the script tag manually in there.
  * `npm run build` does not seem to work out of the box.
  
image
I saw this error, not sure how to resolve it.
  
  * If the build did work, the `bundle.js` that is included in the export is different from the one that would be generated, because it does not include the full JS of dependencies.
  
  
I propose we do the following:

 * In the export, we provide the `index.html` that VizHub generates internally, which _does include_ the `<script>` tags that pull in `bundle.js`, and also the ones that pull in dependencies from UNPKG.
 * In the exported Rollup config, treat dependencies as external, so that when the user runs `npm run build`, it would generate the same lightweight bundle that VizHub would generate.
 
The alternative option would be to go all-in with providing users an exported thing that takes full advantage of NPM and actually bundles installed dependencies into the build. This is a "nice to have", and seems to be more trouble than it's worth. If we did this, we'd need to do a lot more work to make sure:
  * The export should be runnable if you just drop the files into a static file server.
  * The `npm run build` command works in all cases
  * The exported bundle should be the same as the one that `npm run build` would generate, so I guess we'd need to generate it server side or something... This seems like way too much work.
  
So, after this is all working, I think whatever confusion remains can be clarified with a blog post that documents what users can do if they do want to switch to using a Rollup config that bundles dependencies into the build. This is a change that users can do manually if they want to. We just need to provide them something that works out of the box, even if it is a little unconventional (e.g. using UNPKG to pull in dependencies).

### PR DESCRIPTION
**Pull request by stushurik**
Closes datavis-tech/vizhub-issue-tracker/issues/672

### CODE REVIEW

File: packages/useCases/src/interactors/exportVisualization/index.js [hunk 1]
```diff
@@ -49,33 +51,13 @@ export class ExportVisualization {
     sourcemap: true,
     globals: ${globalsJSON}
   },
-  plugins: [nodeResolve(), buble({exclude: ['node_modules/**/*']})]
+  plugins: [buble()]
```
Comment (curran) [inline packages/useCases/src/interactors/exportVisualization/index.js:54]: Interesting. I thought we needed `nodeResolve`. Will test it out.

**Review by curran — APPROVED** [review]
Looks great!

**Review by curran — COMMENTED** [review]
I see a problem.

Before this PR:

image


After this PR:

image

Any ideas what might be the cause of this size increase?

**Review by curran — COMMENTED** [review]
Also I'm seeing some new circular dependency warnings.

image

**Review by curran — COMMENTED** [review]
Strange, when I build this on the staging server (2GB RAM), the build step fails.

image

This has always worked before.

I wonder if JSDOM is a huge library or something? No idea.

**Comment by stushurik** [comment]
@curran jsdom should be required only server side, we need to check if it was bundled into client chunk

**Comment by stushurik** [comment]
@curran I used jsdom cause "querySelector" was used in order to find out whether the bundle script present in dom. xmldom does not provide that method

**Comment by curran** [comment]
Thanks for that info! Good to know.

Yeah I'm confused as to why the bundle size would change at all, as it should indeed only be pulled in on the server side, not bundled with the client.

Investigating...

**Pull request merged**